src/compose/ComposeBox.js

```diff
@@ -217,6 +219,36 @@ class ComposeBox extends PureComponent<Props, State> {
     }
   };
 
+  replaceMessageTextAtRange = (text: string, beg: number, end: number) => {
+    const { message } = this.state;
+    this.setMessageInputValue(message.slice(0, beg) + text + message.slice(end, message.length));
+    return [beg, beg + text.length];
+  };
+
+  insertAttachment = async (attachments: [DocumentPickerResponse]) => {
+    this.setState(({ isUploading }) => ({
+      isUploading: !isUploading,
+    }));
```

Comment:
Hmm, this doesn't seem quite right -- it means that if for whatever reason when this `setState` callback gets run, the value of `isUploading` is already true, it'll get set to *false*.

Probably closer to what you want would just be: `this.setState({ isUploading: true })`. That is, we definitely want to be making the value more true here, and never less true.

Reply:
fixed. using `numUploading` now.